### ISSUE TITLE
Ecs App Load Balancer Create Error

### ISSUE THREAD

**Issue by jebrage** [cloudtools/stacker_blueprints#32]
I'm trying to create a stack as specified below using blueprints.

```
  - name: lbtg-mobile-${env_name}
    class_path: stacker_blueprints.generic.GenericResourceCreator
    variables:
      Class: elasticloadbalancingv2.TargetGroup
      Output: LBTargetGroupId
      Properties:
        Name: lbtg-mobile-${env_name}
        Port: 443
        Protocol: HTTPS
        VpcId: ${output vpc-${env_name}::VpcId}

  - name: appcluster-${env_name}
    class_path: stacker_blueprints.ecs.Cluster

  - name: app-mobile-${env_name}
    class_path: stacker_blueprints.ecs.SimpleECSApp
    variables:
      TaskName: MobileAppTask-${env_name}
      Cluster: ${output appcluster-${env_name}::ClusterArn}
      Image: 70xxxxxxxxx121.dkr.ecr.ap-south-1.amazonaws.com/mobile@sha256:e089xxxxxxxxxxxxx987f407f3c069edf45e5
      CPU: 1024
      Memory: 2048
      AppName: MobileApp-${env_name}
      ContainerPort: 8080
      HostPort: 80
      LoadBalancerTargetGroupArns:
        - ${output lbtg-mobile-${env_name}::LBTargetGroupId}
```

I am facing an error when it is getting created.

> CREATE_FAILED AWS::ECS::Service The target group with targetGroupArn arn:aws:elasticloadbalancing:ap-south-1:70xxxxxx121:targetgroup/lbtg-mobile-stage/36axxxxxx9 does not have an associated load balancer.

**Comment by jebrage** [comment]
A similar issue is faced here https://github.com/hashicorp/terraform/issues/12634

**Comment by phobologic** [comment]
That's because you actually have to have a full load balancer created, though the ecs app only needs to know about the target group arns.

You'll probably want to create both the target group, load balancer, and listener resources - then this should work fine (we actually do this with private blueprints at Modsy - one per each of those types of resources, and it works fine with the open source stacker_blueprint ECS blueprints).

**Comment by phobologic** [comment]
Hey @jebrage - haven't heard back from you, I'm going to go ahead and close this - let me know if my suggestion didn't help.  Thanks!

**Comment by jebrage** [comment]
Hey apology for being late. I got it to work finally.
I configured an Application Load Balancer ie `elasticloadbalancingv2.LoadBalancer` in troposphere, added a target group `elasticloadbalancingv2.TargetGroup` and mapped them both with Listeners in ports 80 and 443 using `elasticloadbalancingv2.Listener`.

I thought the blueprint was adding a load balancer to the template because of this line which is what confused me.

    def generate_load_balancers(self):
        load_balancers = []
        for arn in self.load_balancer_target_group_arns:
            load_balancers.append(
                ecs.LoadBalancer(
                    ContainerName=self.container_name,
                    ContainerPort=self.container_port,
                    TargetGroupArn=arn,
                )
            )
        return load_balancers or NoValue

Anyways adding the Application Load Balancer solved the issue.

I had to use the GenericResourceCreator for the listeners, target groups and the Application Load balancer and its becoming quite messy now as its like one stack just for one listener.

It would be great if we could add that as well to the ECS App or make another extension.
